### PR TITLE
NetworkManager: update to 1.40.0

### PR DESCRIPTION
**Pull request by cinerea0**
#### Testing the changes
- I tested the changes in this PR: **YES**

### CODE REVIEW

File: srcpkgs/NetworkManager/patches/fix-cross-gir.patch [hunk 1]
```diff
@@ -1,10 +1,10 @@
-Reason: Fix crossbuilding GIR
-Upstream: No, not upstreamable since this is very specific to the Void setup
-Index: b/src/libnm-client-impl/meson.build
-===================================================================
+# Reason: Fix crossbuilding GIR
+# Upstream: No, not upstreamable since this is very specific to the Void setup
+diff --git a/src/libnm-client-impl/meson.build b/src/libnm-client-impl/meson.build
+index 46464a6328..27a42f51e6 100644
 --- a/src/libnm-client-impl/meson.build
 +++ b/src/libnm-client-impl/meson.build
```
Comment (paper42) [inline srcpkgs/NetworkManager/patches/fix-cross-gir.patch:6]: what's the reason for this?

**Comment by cinerea0** [comment]
The errors appear to be most commonly caused by appstream not being present. The following configurations have resulted in failure:

* `AppStream` in hostmakedepends
* `appstream-glib` in hostmakedepends and `appstream-glib-devel` in makedepends
* `appstream-glib` and `appstream-glib-devel` in makedepends

**Review by paper42 — COMMENTED** [review]
intltool in hostmakedepends can probably be replaced by gettext

**Comment by paper42** [comment]
> The errors appear to be most commonly caused by appstream not being present. The following configurations have resulted in failure:
> 
>     * `AppStream` in hostmakedepends
> 
>     * `appstream-glib` in hostmakedepends and `appstream-glib-devel` in makedepends
> 
>     * `appstream-glib` and `appstream-glib-devel` in makedepends

the error message is about a polkit policy file, not an appstream file, add polkit-devel to hostmakedepends

File: srcpkgs/NetworkManager/patches/fix-cross-gir.patch (hunk 1 above)
Comment (cinerea0) [reply srcpkgs/NetworkManager/patches/fix-cross-gir.patch:6]: The patch no longer applied due to the meson.build file changing slightly, so I decided to recreate it. That said, the docs option now requires the introspection option, so I'm wondering if there may be a way to use `vopt` to drop the need for this patch altogether.

Comment (paper42) [reply srcpkgs/NetworkManager/patches/fix-cross-gir.patch:6]: this patch applies fine, the other one needed a rebase

File: srcpkgs/NetworkManager/template [hunk 2]
```diff
@@ -35,7 +35,7 @@ license="GPL-2.0-or-later"
 homepage="https://wiki.gnome.org/Projects/NetworkManager"
 changelog="https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/raw/nm-1-38/NEWS"
```
Comment (paper42) [inline srcpkgs/NetworkManager/template:36]:
```suggestion
changelog="https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/raw/nm-1-40/NEWS"
```

**Pull request merged**